src/pages/komodo-defi-framework/api/common_structures/index.mdx

```diff
@@ -499,3 +499,84 @@ The `PagingOptions` object includes options of page selection to consult when lo
+  		"cosmos:irishub-1"
+  	],
+  	"methods": [
+  	  	"cosmos_signDirect",
+  		  "cosmos_signAmino",
+  		  "cosmos_getAccounts"
+  	],
+  	"events": []
+  }
+  ```
+</CollapsibleSection>
+
+### WcSession
+
+The `WcSession` object contains details of active WalletConnect sessions.
+
+| Parameter        | Type    | Description                                                                                                                                                                                                                                                                |
+| ---------------- | ------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| topic            | string  | The session topic hex string, used to identify the session for communications with the external wallet/dapp.                                                                                                                                                               |
+| metadata         | object  | Arbitrary data returned from the external wallet/dapp on connection.                                                                                                                                                                                                       |
+| peer\_pubkey     | string  | TBA                                                                                                                                                                                                                                                                        |
+| pairing\_topic   | string  | The pairing topic hex string, used for the initial external wallet/dapp connection.                                                                                                                                                                                        |
+| namespaces       | object  | Contains the same two [WcConnNs](/komodo-defi-framework/api/common_structures/#wc-conn-ns) objects for Cosmos/EIP155 which were input when initialising the connection via [wc\_new\_connection](/komodo-defi-framework/api/v20-dev/wc_new_connection/#wc-new-connection). |
+| subscription\_id | string  | A hex string to identify the connection's subscription                                                                                                                                                                                                                     |
+| properties       | Null    | TBA                                                                                                                                                                                                                                                                        |
```

Comment:
@borngraced please assist with description here also

Reply:
The below field doesn't really seem relevant to the gui instead should be kept internally, I have updated my PR to exclude this field from get session response.

`peer_pubkey`
`subscription_id`
`properties`

I guess these fields are enough for the gui

```rust
    pub struct SessionRpcInfo {
    pub topic: Topic,
    pub metadata: Metadata,
    pub pairing_topic: Topic,
    pub namespaces: BTreeMap<String, Namespace>,
    pub expiry: u64,
}
```

I'm not sure how `namespaces` can be useful tho but I will leave it.